runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,539 @@
+		_, err = c.dockerClient.VolumeCreate(ctx, volume.CreateOptions{
+			Name: scanName,
+		})
+		if err != nil {
+			return provider.FatalError{
+				Err: fmt.Errorf("failed to create volume. Provider=%s: %w", models.Docker, err),
+			}
+		}
+	}
+
+	// TODO(paralta) Check if volume is empty before adding export content
+	readCloser, err := c.export(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to export target. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Create an ephemeral container to populate volume with export output
+	_, err = c.dockerClient.ImagePull(ctx, "alpine", types.ImagePullOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to pull helper image. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	response, err := c.dockerClient.ContainerCreate(ctx,
+		&container.Config{
+			Image: "alpine",
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: scanName,
+					Target: "/data",
+				},
+			},
+		}, nil, nil, "")
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create helper container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	defer func() {
+		err = c.dockerClient.ContainerRemove(ctx, response.ID, types.ContainerRemoveOptions{})
+		if err != nil {
+			_ = fmt.Errorf("failed to remove helper container. Provider=%s: %w", models.Docker, err)
+		}
+	}()
+	err = c.dockerClient.CopyToContainer(ctx, response.ID, "/data", readCloser, types.CopyToContainerOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to copy data to container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// TODO(paralta) Check if config file exists
+	scanConfigFilePath, err := c.createScanConfigFile(config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scanconfig.yaml file. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.startScan(ctx, scanName, scanConfigFilePath, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to run scan on the scanner container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	return nil
+}
+
+func (c Client) startScan(ctx context.Context, volumeName string, scanConfigFilePath string, config *provider.ScanJobConfig) error {
+	pl, err := c.dockerClient.ImagePull(ctx, config.ScannerImage, types.ImagePullOptions{})
+	if err != nil {
+		return err
+	}
+	_, _ = io.Copy(io.Discard, pl)
+	_ = pl.Close()
+
+	script := fmt.Sprintf(`
+/app/vmclarity-cli \
+--config /tmp/%s \
+--server %s \
+--asset-scan-id %s \
+--output /var/opt/vmclarity
+`, filepath.Base(scanConfigFilePath), config.VMClarityAddress, config.AssetScanID)
+
+	// TODO(paralta) Check if container already created
+	resp, err := c.dockerClient.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image:      config.ScannerImage,
+			Entrypoint: []string{"sh", "-c", script},
+		},
```

Comment:
Can we have this a standard command rather than an entrypoint?
```go
&container.Config{
	Image: config.ScannerImage,
	Cmd:   []string{"/app/vmclarity-cli", "--config", fmt.Sprintf("/tmp/%s", filepath.Base(scanConfigFilePath), // add the rest
},

```

Reply:
will change this